inconspiquous/dialects/qu.py

```diff
@@ -51,6 +67,37 @@ def num_qubits(self) -> int:
         return 1
 
 
+@irdl_op_definition
+class FromBitsOp(IRDLOperation):
+    name = "qu.from_bits"
+    qubits = var_operand_def(BitType)
+    reg = result_def(QReg)
+
+
+@irdl_op_definition
+class ToBitsOp(IRDLOperation):
+    name = "qu.to_bits"
+    reg = operand_def(QReg)
+    qubits = var_result_def(BitType)
+
+
+@irdl_op_definition
+class CombineOp(IRDLOperation):
+    name = "qu.combine"
+    reg1 = operand_def(QReg)
+    reg2 = operand_def(QReg)
+    res = result_def(QReg)
```

Comment:
Trying to get the constraints correct here (and for `SplitOp`) would require addition of integer constraints, which we haven't added upstream yet, so I'd be ok with leaving the constraints of these two operations as they are here.